ehrapy/api/encode/encode.py

```diff
@@ -15,13 +16,55 @@
 
 
 class Encoder:
-    """The main encoder for the initial read AnnData object providing various encoding solutions for
+    """The main encoder for the initial read MuData or AnnData object providing various encoding solutions for
     non numerical or categorical data"""
 
     available_encodings = {"one_hot_encoding", "label_encoding", "count_encoding"}
 
     @staticmethod
     def encode(
+        data: Union[AnnData, MuData],
+        autodetect: Union[bool, Dict] = False,
+        encodings: Union[Dict[str, Dict[str, List[str]]], Dict[str, List[str]]] = None,
+    ) -> Optional[AnnData]:
+        """Driver function for encoding"""
+        if isinstance(data, AnnData):
+            # basic type checking for passed parameters when encoding a single AnnData object
+            if not Encoder._check_anndata_input_type(autodetect, encodings):
+                sys.exit(1)
```

Comment:
Again, please NO sys.exits!!!
This is not a CLI application! Throw an exception and/or a red rich print with a return.